parser/src/display_hint.rs

```diff
@@ -67,7 +67,15 @@ impl DisplayHint {
                         name: bitflags_name.into(),
                         package: package.into(),
                         disambiguator: disambiguator.into(),
-                        crate_name: crate_name.into(),
+                        crate_name: Some(crate_name.into()),
+                    });
+                }
+                [bitflags_name, package, disambiguator] => {
+                    return Some(DisplayHint::Bitflags {
+                        name: bitflags_name.into(),
+                        package: package.into(),
+                        disambiguator: disambiguator.into(),
+                        crate_name: None,
```

Comment:
This variant became a little more complicated due to the fact that `crate_name` needs to be an `Optional`.
I chose to structure it a little bit different to simplify the code. Do you like this variant?
```
            if parts.len() < 3 || parts.len() > 4 {
                return Some(DisplayHint::Unknown(s.into()));
            }
            return Some(DisplayHint::Bitflags {
                name: parts[0].into(),
                package: parts[1].into(),
                disambiguator: parts[2].into(),
                crate_name: parts.get(3).map(|&s| s.to_string()),
            });
```

The part `|| parts.len() > 4` could be omitted, so the code would ignore additional components. That way, it would be possible to add further information in the future, without a breaking change.
However I'm not sure if such an extension is likely, or if it's better to have a stricter parser.